v1/microk8s-ha.yaml

```diff
@@ -0,0 +1,180 @@
+# This blueprint creates a 3-node microk8s ha cluster for juju (nested multipass).
+# A microk8s controller and an empty model are created as part of the cloud-init script
+# so you can `juju deploy` right away.
+# Unlike "charm-dev", this blueprint does not install development tools.
+#
+# References:
+# - https://discuss.kubernetes.io/t/microk8s-in-lxd/11520
+
+description: A 3-node microk8s cluster for juju
+version: latest
+
+runs-on:
+  - x86_64
+  - arm64
+  - s390x
+
+instances:
+  microk8s-ha:
+    image: 22.04
+    limits:
+      min-cpu: 8
+      min-mem: 16G
+      min-disk: 100G
+    timeout: 1800
+    cloud-init:
+      vendor-data: |
+        package_update: true
+
+        packages:
+        - jq
+        - kitty-terminfo
+
+        snap:
+          commands:
+          - snap install lxd --channel=5.21/stable
```

Comment:
Versioning on components can get outdated quickly.
Thoughts on not setting a channel (and trusting default tracks)?

Reply:
You're right, and indeed the charm-dev blueprint needed some updates from time to time.
But the updates weren't just bumping the channel; it involved changes related to juju moving from classically to strictly confined, and changes in how lxd is initialized.

Ideally, we would have a renovate-like solution for bumping versions hardcoded in a blueprint.
In its absence I believe pinning to well-tested, manually verified version combos would give better value to users.